scylla/tests/integration/tablets.rs

```diff
@@ -0,0 +1,601 @@
+                return running_proxy;
+            }
+
+            let ks = unique_keyspace_name();
+
+            /* Prepare schema */
+            prepare_schema(&session, &ks, "t", TABLET_COUNT).await;
+
+            let tablets = get_tablets(&session, &ks, "t").await;
+
+            let prepared = session
+                .prepare(format!(
+                    "INSERT INTO {}.t (a, b, c) VALUES (?, ?, 'abc')",
+                    ks
+                ))
+                .await
+                .unwrap();
+
+            let value_lists = calculate_key_per_tablet(&tablets, &prepared);
+
+            let (feedback_txs, mut feedback_rxs): (Vec<_>, Vec<_>) = (0..3)
+                .map(|_| mpsc::unbounded_channel::<(ResponseFrame, Option<TargetShard>)>())
+                .unzip();
+            for (i, tx) in feedback_txs.iter().cloned().enumerate() {
+                running_proxy.running_nodes[i].change_response_rules(Some(vec![ResponseRule(
+                    Condition::ResponseOpcode(ResponseOpcode::Result)
+                        .and(Condition::not(Condition::ConnectionRegisteredAnyEvent)),
+                    ResponseReaction::noop().with_feedback_when_performed(tx),
+                )]));
+            }
+
+            // At first, driver has no info about tablets. When it has no info for given tablet,
+            // DefaultPolicy will fall back to non-shard-aware routing, so if we send a lot of
+            // queries then we should get at least one tablet feedback. We are
+            // running with 3-node cluster and keyspace has replication factor 2,
+            // so even if each node has only one shard there must exist at least
+            // one non-replica node.
+            // Initially there are not tablets, so LBP will fallback to token ring.
+            // Tablet replicas and toklen ring replicas may overlap sometimes,
+            // so we can't guarantee a reply for each tablet, until we get any,
+            // so that LBP knows that table is a tablet table.
+            // Solution is to iterate over the tablets twice. First iteration
+            // will guarantee that we receive some feedback (in practice: pretty much
+            // all feedbacks), second iteration receives missing ones.
+            let mut total_tablets_with_feedback = 0;
+            for values in value_lists.iter().chain(value_lists.iter()) {
+                info!(
+                    "First loop, trying key {:?}, token: {}",
+                    values,
+                    prepared.calculate_token(&values).unwrap().unwrap().value()
+                );
+                try_join_all((0..100).map(|_| async { session.execute(&prepared, values).await }))
+                    .await
+                    .unwrap();
+                let feedbacks: usize = feedback_rxs.iter_mut().map(count_tablet_feedbacks).sum();
+                if feedbacks > 0 {
+                    total_tablets_with_feedback += 1;
+                }
```

Comment:
When are we going to have `feedbacks > 1`? Are we sure all those feedbacks come from the same tablet?